linker/shared/src/main/scala/org/scalajs/linker/analyzer/Analyzer.scala

```diff
@@ -423,6 +423,7 @@ private final class Analyzer(config: CommonPhaseConfig,
     var areInstanceTestsUsed: Boolean = false
     var isDataAccessed: Boolean = false
     var isAnyStaticFieldReachable: Boolean = false
+    var isReferenced: Boolean = false
```

Comment:
`isReferenced` is never actually used. I believe that it can be removed, and simply doing the `lookupClass()` is enough.

That also means that the calls to `reference()` are superfluous.